lib/hal_client/representation.rb

```diff
@@ -294,6 +315,40 @@ def raw
 
     MISSING = Object.new
 
+    def flatten_section(section_hash)
+      result = Array.new
+
+      section_hash.each_pair do |rel, rel_data|
+        if rel_data.is_a?(Array)
```

Comment:
Rather than type checking could let's use the `[rel_data].flatten`. As in:
```ruby
section_hash
  .each_pair
  .flat_map { |rel, some_link_info| 
    [some_link_info].flatten
      .map { |a_link_info| {rel: rel, data: a_link_info} }
  }
```